pkg/destroy/gcp/disk.go

```diff
@@ -10,8 +10,37 @@ import (
 	"google.golang.org/api/googleapi"
 )
 
+const (
+	maxGCEPDNameLength    = 63
+	estimatedPVNameLength = 40
+	// Removing an extra value (-1) for the "-" separated between the storage name and the pv name
+	storageNameLength = maxGCEPDNameLength - estimatedPVNameLength - 1
+)
+
+// formatClusterIDForStorage will format the Cluster ID as it will be used for creating
+// GCE PDs. The maximum length is 63 characters, and can end with "-dynamic".
+// https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/util.go, GenerateVolumeName()
+func (o *ClusterUninstaller) formatClusterIDForStorage() string {
+	storageName := o.ClusterID + "-dynamic"
+	return storageName[:storageNameLength]
```

Comment:
Do the disks whos names are truncated still end in "-dynamic", and if so, does this handle that?

Reply:
yes, this uses the exact formula that kubernetes uses for their storage naming convention. I added a link in the comment above the function so that it is known where that information came from (and in the event of a change).